### DISCUSSION TITLE
Use with maven behind a proxy

### DISCUSSION THREAD

**Discussion by eneufeld** [eclipse/dash-licenses#149] [Q&A]
Hi,

thank you for the great project.
We want to use this behind a proxy but unfortunately the connection does not work.
I know the maven proxy settings work as I added the `download-maven-plugin` and it reaches eg `https://www.eclipse.org/legal/licenses.json`.
Looking at the source code I think a setting the proxy is missing in https://github.com/eclipse/dash-licenses/blob/master/core/src/main/java/org/eclipse/dash/licenses/http/HttpClientService.java .

Did I miss something in my config?
Thank you for your help.

Best,
Eugen

**Comment by waynebeaton** [comment]
I don't have any recent experience with configuring for proxy servers, but my understanding is that the Java 11 HTTP Client that the implementation uses can be configured for proxy via system properties. See [here](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/doc-files/net-properties.html#Proxies).

I believe that if you add `-Dhttps.proxyHost=... -Dhttps.proxyPort=...` to the command-line, you'll accomplish what you need.

Please confirm whether or not this works for you.

**Reply by eneufeld** [reply]
I think I tried this and it didn't work. But looking through the source code of OpenJDK for the HTTPClient it actually should work. I try again and comeback when I know more.
Thank you for the quick reply.

**Reply by eneufeld** [reply]
Ok so I checked again. Yes the approach would work.
But in the case of a proxy that needs authentication you don't want to pass them as clear text parameters especially when you use the dash-licenses as a maven plugin.
In this case I would like to use the settings.xml as described here: https://maven.apache.org/guides/mini/guide-encryption.html

Are there any ideas on how to achieve this?

**Reply by waynebeaton** [reply]
Frankly... no idea.

If you can please capture what you learn while solving this, we can add it to the documentation.

**Reply by eneufeld** [reply]
Hi @waynebeaton , 
We opened a PR to handle the proxy settings.
I hope this is also in your interest.
